### PR TITLE
"Mathable"

### PR DESCRIPTION
**Pull request by stephentyrone**

### CODE REVIEW

File: proposals/0000-mathable.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+let y = Float.Math.exp(x) // 2.7182817
+```
+Additionally, introduce a new *module*, `Math`, that provides the "customary" free 
+functions:
+```swift
+import Math
+let z = exp(x) // 2.7182817
+```
+By doing this, the free functions are not defined in the global namespace unless
+this module is explicitly improted, but the functionality is always available via the static
```
Comment (benrimmington) [inline proposals/0000-mathable.md]:
```suggestion
this module is explicitly imported, but the functionality is always available via the static
```

File: proposals/0000-mathable.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+Specifically, we will provide two overloads: `pow(_ x: T, _ y: T) -> T` and 
+`pow(_ x: T, _ n: Int) -> T`, corresponding to the IEEE-754 "powr" and "pown"
+operations.
+- There are two things that are weird about `atan2`. First, the argument order is trap.
+The signature is `atan2(_ y: T, _ x: T) -> T`, and it computes `atan(y/x)` with
+some fixup around how the sign is handled, which makes it equivalent to getting the
+angle of the polar representation of the vector `(x, y)`. Second, it doesn't really make
+sense for complex types, which Mathable is intended to support; I don't know of
+another language that provides it on complex `T`. We can define it as `atan(y/x)` in
+those cases, but it's slightly weird. Ultimately, a Swifiter interface should be provided
```
Comment (benrimmington) [inline proposals/0000-mathable.md]:
```suggestion
those cases, but it's slightly weird. Ultimately, a Swiftier interface should be provided
```

**Comment by stephentyrone** [comment]
> `Mathable` => `Mathematical` ?

Mathematical isn't really satisfactory, either. All programming is "mathematical". To some extent this applies to "Mathable" too, of course. If we want to be precise, we would do something like "HasTranscendentalFunctions" or whatever, but that's pretty gross to have as a generic constraint; we want this to be semi-approachable for non-experts.

**Comment by rjmccall** [comment]
We'd like to put this up for review on the 11th.  Let me know when you've crossed all your `t`s and I'll merge and assign an SE.

**Comment by stephentyrone** [comment]
@rjmccall I'll give folks until EOD today to make any straggling suggestions for alternatives considered.

**Comment by tkremenek** [comment]
I will merge this on @rjmccall's behalf and kick off the review with John as review manager.

**Comment by rjmccall** [comment]
Thanks, @tkremenek.  It was a rough day today.

**Pull request merged**